### ISSUE TITLE
feat: ROCKET/MiniRocket feature extraction for clustering

### ISSUE THREAD

**Issue by thomaschi78** [drumtorben/polars-ts#95]
## Context

Random convolutional kernels (ROCKET/MiniRocket) transform time series into fixed-length feature vectors extremely fast. A 2024 DAMI paper showed these features achieve **state-of-the-art clustering results** when fed into standard clustering algorithms, outperforming many elastic-distance methods while being orders of magnitude faster.

**Tier: 2 (moderate effort)** — new feature extraction module, then cluster in feature space with existing or sklearn methods.

**References:**
- Dempster, A. et al. (2020). *ROCKET: Exceptionally fast and accurate time series classification using random convolutional kernels*. DMKD.
- Dempster, A. et al. (2021). *MiniRocket: A very fast (almost) deterministic transform for time series classification*. KDD.
- Holder, C. et al. (2024). *Random Convolutional Kernels for Time Series Clustering*. DAMI.

## Proposed API

```python
from polars_ts import rocket_features, minirocket_features

# Extract features
features = rocket_features(df, n_kernels=500)
# Returns: pl.DataFrame with [unique_id, rocket_0, rocket_1, ..., rocket_999]
# (500 kernels × 2 features each = 1000 features)

features = minirocket_features(df, n_kernels=500)
# Deterministic variant, ~75x faster

# Then cluster in feature space
from sklearn.cluster import KMeans
X = features.drop("unique_id").to_numpy()
labels = KMeans(n_clusters=3).fit_predict(X)
```

### Task 1: Implement ROCKET feature extraction
- **File:** `polars_ts/features/rocket.py` (new)
- Generate random convolutional kernels (varying length, dilation, bias)
- Apply each kernel to each series → extract PPV (proportion of positive values) and max
- Pad/truncate series to equal length
- **Acceptance criteria:**
  - [ ] Returns DataFrame with `[id_col, feature_0, ..., feature_{2*n_kernels-1}]`
  - [ ] Features are deterministic with seed
  - [ ] Handles variable-length series
- **Scope:** Medium (new algorithm, but well-documented)

### Task 2: Implement MiniRocket variant
- Fixed kernel lengths (9), mostly deterministic
- Significantly faster (no random sampling of lengths/dilations)
- **Scope:** Medium

### Task 3: Register, test, add to notebook 07
- **Scope:** Small

## Verification
```bash
uv run pytest tests/features/test_rocket.py -v
```
